cmd/cosign/cli/verify_attestation.go

```diff
@@ -166,10 +200,92 @@ func (c *VerifyAttestationCommand) Exec(ctx context.Context, args []string) (err
 		co.VerifyBundle = false
 
 		verified, err := cosign.Verify(ctx, ref, co)
+
 		if err != nil {
 			return err
 		}
 
+		var validationErrors []error
+		for _, vp := range verified {
+			var payloadData map[string]interface{}
+			err := json.Unmarshal(vp.Payload, &payloadData)
+			if err != nil {
+				return fmt.Errorf("unmarshal payload data: %w", err)
+			}
+
+			// sanity checks
+			if val, ok := payloadData["payloadType"]; ok {
+				// we need to check only given type from the cli flag
+				// so we are skipping other types
+				if predicateURI != val {
+					continue
+				}
+			} else {
+				return fmt.Errorf("could not find 'payloadType' in payload data")
+			}
+
+			var decodedPayload []byte
+			if val, ok := payloadData["payload"]; ok {
+				decodedPayload, err = base64.StdEncoding.DecodeString(val.(string))
+				if err != nil {
+					return fmt.Errorf("could not decode 'payload': %w", err)
+				}
+			} else {
+				return fmt.Errorf("could not find 'payload' in payload data")
+			}
+
+			var payload []byte
+			switch c.PredicateType {
+			case predicateCustom:
+				var cosignStatement in_toto.Statement
+				if err := json.Unmarshal(decodedPayload, &cosignStatement); err != nil {
+					return fmt.Errorf("unmarshal CosignStatement: %w", err)
+				}
+				payload, err = json.Marshal(cosignStatement.Predicate)
+				if err != nil {
+					return fmt.Errorf("error when generating CosignStatement: %w", err)
+				}
+			case predicateLink:
+				var linkStatement in_toto.LinkStatement
+				if err := json.Unmarshal(decodedPayload, &linkStatement); err != nil {
+					return fmt.Errorf("unmarshal LinkStatement: %w", err)
+				}
+				payload, err = json.Marshal(linkStatement.Predicate)
+				if err != nil {
+					return fmt.Errorf("error when generating LinkStatement: %w", err)
+				}
+			case predicateSlsa:
+				var slsaProvenanceStatement in_toto.ProvenanceStatement
+				if err := json.Unmarshal(decodedPayload, &slsaProvenanceStatement); err != nil {
+					return fmt.Errorf("unmarshal ProvenanceStatement: %w", err)
+				}
+				payload, err = json.Marshal(slsaProvenanceStatement.Predicate)
+				if err != nil {
+					return fmt.Errorf("error when generating ProvenanceStatement: %w", err)
+				}
+			case predicateSpdx:
+				var spdxStatement in_toto.SPDXStatement
+				if err := json.Unmarshal(decodedPayload, &spdxStatement); err != nil {
+					return fmt.Errorf("unmarshal SPDXStatement: %w", err)
+				}
+				payload, err = json.Marshal(spdxStatement.Predicate)
+				if err != nil {
+					return fmt.Errorf("error when generating SPDXStatement: %w", err)
+				}
+			}
+			if err := cue.ValidateJSON(payload, c.Policies.EntryPoints); err != nil {
+				validationErrors = append(validationErrors, err)
+			}
+		}
+
+		if len(validationErrors) > 0 {
+			fmt.Println("Some errors occurred during the validation:")
+			for _, v := range validationErrors {
+				_, _ = fmt.Fprintf(os.Stderr, "- %v\n", v)
```

Comment:
suggestion: use fmt.Printf( instead of Fprintf

Reply:
Do you mean printing errors to the STDOUT?